MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -0,0 +1,1080 @@
+                }
+            }
+        }
+        
+        // now, we get all the forces that qualify as "lances", and filter out those that are
+        // deployed to a scenario and not in a track already
+        for(int key : campaign.getLances().keySet()) {
+            Force force = campaign.getForce(key);
+            if(force != null && 
+                    !force.isDeployed() && 
+                    (force.getScenarioId() <= 0) &&
+                    !forcesInTracks.contains(force.getId())) {
+                retVal.add(force.getId());
+            }
+        }
+        
+        return retVal;
+    }
+    
+    /**
+     * This is a list of all force IDs for forces that can be deployed to a scenario in the given force template
+     * a) have not been assigned to a track
+     * b) are combat-capable
+     * c) are not deployed to a scenario
+     */
+    public static List<Integer> getAvailableForceIDs(int unitType, Campaign campaign, StratconTrackState currentTrack, 
+            boolean reinforcements) {
+        List<Integer> retVal = new ArrayList<>();
+        
+        Set<Integer> forcesInTracks = new HashSet<>();
+        // assemble a set of all force IDs that are currently assigned to tracks that are not this one
+        for (Contract contract : campaign.getActiveContracts()) {
+            if (contract instanceof AtBContract) {
+                for (StratconTrackState track : ((AtBContract) contract).getStratconCampaignState().getTracks()) {
+                    if ((track != currentTrack) || !reinforcements) {
+                        forcesInTracks.addAll(track.getAssignedForceCoords().keySet());
+                    }
+                }
+            }
+        }
+        
+        for (int key : campaign.getLances().keySet()) {
+            Force force = campaign.getForce(key);
+            
+            if (force == null) {
+                continue;
+            }
+            
+            int primaryUnitType = force.getPrimaryUnitType(campaign);
+            if (!force.isDeployed() && 
+                    (force.getScenarioId() <= 0) &&
+                    !force.getUnits().isEmpty() &&
+                    !forcesInTracks.contains(force.getId()) &&
+                    forceCompositionMatchesDeclaredUnitType(primaryUnitType, unitType, reinforcements)) {
+                retVal.add(force.getId());
+            }
+        }
+        
+        return retVal;
+    }
+    
+    /**
+     * Returns a list of individual units eligible for deployment in scenarios run by "Defend" lances
+     * @param campaign
+     * @return List of unit IDs.
+     */
+    public static List<Unit> getEligibleDefensiveUnits(Campaign campaign) {
+        List<Unit> retVal = new ArrayList<>();
+        
+        for(Unit u : campaign.getUnits()) {
+            // "defensive" units are infantry, battle armor and (Weisman help you) gun emplacements
+            if(((u.getEntity().getUnitType() == UnitType.INFANTRY) || 
```

Comment:
This should probably filter on mothballed units and units without a commander or something like that.

Maybe add something to `Hangar`:
```java
    public void forEachActiveUnit(Consumer<Unit> consumer) {
        units.forEach((id, unit) -> {
            if (!unit.isMothballed() && !unit.isUnmanned()) {
                consumer.accept(unit));
            }
        });
    }
```

Reply:
I'm not trying to do anything to the units, so I'm not sure why I'd use this particular pattern. But point taken about mothballed/uncrewed units (although gun emplacements are weird and sometimes can be deployed without crew in a MekHQ context - "as autoturrets")